geopandas/sindex.py

```diff
@@ -32,6 +32,20 @@ def _get_sindex_class():
     )
 
 
+class UnknownPredicateError(ValueError):
+    def __init__(self, predicate, valid_predicates=None, message=None):
+        valid_predicates = valid_predicates or set()
+        message = message or "Predicate {} is not valid; must be one of {}".format(
+            predicate,
+            valid_predicates,
+        )
+        super().__init__(message)
```

Comment:
This is an attempt to make a more cohesive error message across implementations, but we could just go with `UnknownPredicateError(ValueError): pass`.

Reply:
I am personally not a big fan of custom error classes, so I'd prefer to raise `ValueError`. Also, if you pass unsupported predicate it is not _unknown_ but not implemented. So maybe `NotImplementedError`?